addons/point_of_sale/static/tests/tours/product_screen_tour.js

```diff
@@ -413,3 +413,30 @@ registry.category("web_tour.tours").add("AutofillCashCount", {
             ProductScreen.cashDifferenceIs(0),
         ].flat(),
 });
+
+registry.category("web_tour.tours").add("ProductSearchTour", {
+    checkDelay: 50,
+    steps: () =>
+        [
+            Chrome.startPoS(),
+            Dialog.confirm("Open Register"),
+            ProductScreen.searchProduct("Test Product"),
+            ProductScreen.productIsDisplayed("Test Product 1"),
+            ProductScreen.productIsDisplayed("Test Product 2"),
+            ProductScreen.searchProduct("Test Produt"), // typo to test the fuzzy search
+            ProductScreen.productIsDisplayed("Test Product 1"),
+            ProductScreen.productIsDisplayed("Test Product 2"),
+            ProductScreen.searchProduct("1234567890123"),
+            ProductScreen.productIsDisplayed("Test Product 1"),
+            ProductScreen.productIsNotDisplayed("Test Product 2"),
```

Comment:
```suggestion
            negateStep(ProductScreen.productIsDisplayed("Test Product 2")),
```
The test currently failed on this step which is probably random.

I think it's more correct to use `negateStep` because the step will wait for the disappearance of the element when it's present in the screen. With that, you can remove the tour step util you introduced.

Reply:
When ever you introduce a test that relies on `run`, try running it multiple time locally. If it randomly fails, then it's not correct.